src/vm/arm64/asmhelpers.asm

```diff
@@ -526,7 +530,18 @@ UpdateCardTable
         strb     w16, [x12, x15]
 
 #ifdef FEATURE_MANUALLY_MANAGED_CARD_BUNDLES
-#error Need to implement for ARM64
+        ; Check if we need to update the card bundle table
+        ldr      x12, wbs_card_bundle_table
+
+        ; x15 := offset within card bundle table
+        lsr      x15, x14, #21
+
+        ldrb     w16, [x12, x15]
+        cmp      w16, 0xFF
+        beq      Exit
+
+        mov      x16, 0xFF
+        strb     w16, [x12, x15]
```

Comment:
This code is trashing x16 which is not mentioned in the comment above as being trashed. Should be changed.  Why not use x12 like Linux.

Reply:
If you absolutely need x16, you will need to adjust the register reserves for the JIT so it doesn't use during a write barrier.

Reply:
I did it this way because the code updating the card table immediately above already trashed x16. We can follow the Linux example though and adjust the code to just trash x12 and x15.

Reply:
I don't think we have officially supported Arm64 on Windows, so that may be the issue.

The JIT is depending on this here. 

https://github.com/dotnet/coreclr/blob/ad54f6e03e77c38482d8538550f1bc13b3267015/src/jit/target.h#L1044-L1081

We should probably just fix it.

Reply:
I changed it to be consistent with Linux and trash only x12 and x15. 

I'm actually not sure the JIT is depending on it - I see both RBM_CALLEE_TRASH_WRITEBARRIER and RBM_CALLEE_TRASH_WRITEBARRIER_BYREF include RBM_CALLEE_TRASH_NOGC which includes RBM_IP0 which is x16. So judging from that we should be ok even trashing x16, but it's no doubt safer and less confusing to just make the Linux and Windows versions behave the same.